src/ansys/geometry/core/designer/design.py

```diff
@@ -759,3 +762,35 @@ def __read_existing_design(self) -> None:
         self._grpc_client.log.debug(f"SharedTopologyTypes set: {num_created_shared_topologies}")
 
         self._grpc_client.log.debug(f"\nSuccessfully read design in: {end - start} s")
+
+    def _update_design_inplace(self) -> None:
+        """
+        Update the design to align with the server side.
+
+        Notes
+        -----
+        This method is used to update the design inside repair tools.
+        Its usage is not recommended for other purposes.
+        """
+        # Clear all the existing information
+        #
+        # TODO: This might go out of sync with the __read_existing_design method
+        #       if the latter is updated and this method is not. Ensure that
+        #       the two methods are in sync.
+        #
+        self._components = []
+        self._bodies = []
+        self._materials = []
+        self._named_selections = {}
+        self._coordinate_systems = {}
+
+        # Get the previous design id
+        previous_design_id = self._design_id
```

Comment:
@RobPasMue , so why this id check needed? @b-matteo

Reply:
This will be needed for future cases... Not only repair tools in which the ID is kept the same.